apps/opencs/view/render/cell.cpp

```diff
@@ -169,6 +169,8 @@ CSVRender::Cell::Cell (CSMWorld::Data& data, osg::Group* rootNode, const std::st
 {
     std::pair<CSMWorld::CellCoordinates, bool> result = CSMWorld::CellCoordinates::fromId (id);
 
+    mTerrainStorage = new TerrainStorage(mData);
```

Comment:
I haven't found `delete mTerrainStorage`. Please use `std::unique_ptr<TerrainStorage>` type.

Reply:
Hmm.. It seems this pointer structure is built deep into OpenMW-CS code. I'm unable to simply change this to std::unique_ptr:

```
/home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/view/render/cell.cpp: In member function ‘void CSVRender::Cell::updateLand()’:
/home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/view/render/cell.cpp:137:83: error: no matching function for call to ‘Terrain::TerrainGrid::TerrainGrid(osg::ref_ptr<osg::Group>&, osg::ref_ptr<osg::Group>&, std::__shared_ptr<Resource::ResourceSystem, __gnu_cxx::_S_atomic>::element_type*, std::unique_ptr<CSVRender::TerrainStorage>&, CSVRender::Mask)’
  137 | Data.getResourceSystem().get(), mTerrainStorage, Mask_Terrain));
      |                                                              ^

In file included from /home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/view/render/cell.cpp:12:
/home/nelsson/Koodi/openmw_code/transientlandediting/openmw/./components/terrain/terraingrid.hpp:17:9: note: candidate: ‘Terrain::TerrainGrid::TerrainGrid(osg::Group*, osg::Group*, Resource::ResourceSystem*, Terrain::Storage*, int, int, int)’
   17 |         TerrainGrid(osg::Group* parent, osg::Group* compileRoot, Resource::ResourceSystem* resourceSystem, Storage* storage, int nodeMask, int preCompileMask=~0, int borderMask=0);
      |         ^~~~~~~~~~~
/home/nelsson/Koodi/openmw_code/transientlandediting/openmw/./components/terrain/terraingrid.hpp:17:117: note:   no known conversion for argument 4 from ‘std::unique_ptr<CSVRender::TerrainStorage>’ to ‘Terrain::Storage*’
   17 | pileRoot, Resource::ResourceSystem* resourceSystem, Storage* storage, int nodeMask, int preCompileMask=~0, int borderMask=0);
      |                                                     ~~~~~~~~~^~~~~~~
```

I'm trying hard not to change the Terrain::Storage or component-level, as it might affect the engine.

Reply:
Ok, it's not so simple. Object is deleted in other place.

* Starting [here](https://github.com/unelsson/openmw/blob/transientlandshapeedit/apps/opencs/view/render/cell.cpp#L136-L137) it's passed to `Terrain::TerrainGrid` [constructor](https://github.com/unelsson/openmw/blob/transientlandshapeedit/components/terrain/terraingrid.cpp#L22).
* Then to `Terrain::World` [constructor](https://github.com/unelsson/openmw/blob/transientlandshapeedit/components/terrain/world.cpp#L18) and stored into `Terrain::World::mStorage`.
* And in `Terrain::World` [destructor](https://github.com/unelsson/openmw/blob/transientlandshapeedit/components/terrain/world.cpp#L66) it is deleted.

I'd suggest to leave as is.